### ISSUE TITLE
Multiple Recordsets in stored procedure

### ISSUE THREAD

**Issue by tidhar-ziv** [typeorm/typeorm#5051]
We are using typeorm to execute a stored procedure in a legacy mssql DB. The procedure has multiple selects, which causes multiple recordsets to be return, and we are interested in the last one, which is the final result of the stored procedure.

The result from the mssql driver contains both recordsets array (in a recordsets property), and the first recordset (in a dedicated recordset property) (see: https://github.com/tediousjs/node-mssql/blob/624219ab0c945d57b72b0d9a949695ff54feaba5/lib/base.js#L1457) 

Typeorm's sqlserver query runner returns only the recordset property (see: https://github.com/typeorm/typeorm/blob/c87e4e7ad92023a1914a6a1778d683ba8f23c5ab/src/driver/sqlserver/SqlServerQueryRunner.ts#L233), so we are getting the first recordset instead of the last.

Is there a way to change this? If not, wouldn't it be more logical to return the last recordset?

Thanks!

**Comment by tidhar-ziv** [comment]
@dhruvaprajapati unfortunately no

**Comment by briandiaz** [comment]
Hey guys,

Any news on this topic?

**Comment by BerenLongstreet** [comment]
This is really frustrating me. I know it is supported by the underlying driver, and also supported by Sequelize.

I can see the other recordset in the internal closures of the toTable method of the query result, but I cannot get to it.

Does anyone have an idea to fix or work around this?

**Comment by pezzetti** [comment]
Sadly, the workaround so far is to use mssql npm package to call the stored procedure directly and manipulate the recordset

**Comment by BerenLongstreet** [comment]
Understood. Is there a quick way to get to the underlying typeorm Driver Connection object through the Entity manager object, without having to create a separate mssql connection?
________________________________

**Comment by diego-forero** [comment]
Any updates on this?

**Comment by BerenLongstreet** [comment]
I made an update on my personal copy to allow this functionality (just a couple of lines I think) and it works great, but I haven't had the time (or the space on my laptop) to set up the required test environment to run the full unit test suite. I'd be happy to send my updates to someone else who can make a PR to get it into the codebase...

Let me know...
Beren Longstreet
________________________________

**Comment by lboonekamp** [comment]
I'm working on a PR for this now as I discovered the same issue with an SP I was executing. How are you executing your SP? Just using query()? I don't think there's another way, so just confirming

**Comment by BerenLongstreet** [comment]
Yep, just using query()... I am still using patch-package to use my forked logic, but it's literally like 3 lines. Let me know what you come up with.
